### ISSUE TITLE
PDF: Render dynamic user data into pdf

### ISSUE THREAD

**Issue by rosibaj** [icgc-argo/dac-ui#128]
- get the app data for this directly from the api (so that part of the pdf won’t rely on what the ui is doing). 
- render the state of the users selections on the pdf template.

**Comment by anncatton** [comment]
@rosibaj @kcullion Known issues from this PR:
- `IT Agreements` and `Data Access Agreements` sections are missing some of the grey borders in the web ui
- the loading state for the PDF download should not resize the button (this happens because the button text changes)
- the PDF download button text is hard coded to `Draft` and does not reflect the current app state

**Comment by kcullion** [comment]
thanks @anncatton. do you suggest we make tickets for all of these or will you be fixing as part of this ticket?

**Comment by anncatton** [comment]
I just added this as a comment on the ticket so you and Rosi are aware. I don't think any new tickets should be made until you've had a chance to QA everything on this ticket, and then we can go from there.

**Comment by kcullion** [comment]
@anncatton @rosibaj I put Ann's second and third point in this new ticket: https://github.com/icgc-argo/dac-ui/issues/181
I put the fourth item in this ticket so we remember to test it: https://github.com/icgc-argo/dac-ui/issues/175 

Rosi please review that third item about the signed pdf download - what was decided - are we zipping or including ethics letter in pdf?

**Comment by kcullion** [comment]
@anncatton Some more feedback 

- [ ] fix the url, change the wording and remove the dash from ICGC-DACO `DACO-70 created for <applicant title><applicant first name><applicant last name> by ICGC DACO (https://daco.icgc-argo.org)` and if there is no applicant name filled in yet, then include  <applicant name> in the string

**Comment by rosibaj** [comment]
Tested Jul 07, 2021: 
- [ ] some ui issues with the publication display
![image](https://user-images.githubusercontent.com/15824252/124793966-83e78a80-df1c-11eb-8d8a-54b2487e1a05.png)

- [ ] lay summary is not being included and should be 
![image](https://user-images.githubusercontent.com/15824252/124794095-a7aad080-df1c-11eb-8ba1-ae4c412d52de.png)
![image](https://user-images.githubusercontent.com/15824252/124794038-98c41e00-df1c-11eb-99d0-96a9b1748280.png)


@kcullion the zip package includes the ethics letter when it is signed. that should be delivered fine by the api rn

**Comment by kcullion** [comment]
- [ ] In draft, if the user hasn't filled in all or any publication urls, still display 3 blank publication url fields. That's what we're doing throughout the pdf. 
![image](https://user-images.githubusercontent.com/18424138/124800748-00ca3280-df24-11eb-9f0a-22d5c7b4452d.png)

- [ ] I hadn't filled out the form yet and the ethics checks are both checked off. these should be blank
![image](https://user-images.githubusercontent.com/18424138/124800923-3707b200-df24-11eb-9f04-692fdba2b319.png)

- [ ] The names on the pdf should always include all of these field values (if they exist) <title><first name><middle name><last name><suffix>. For Applicant, Inst rep and collaborators.  I entered all of those values and see this:
![image](https://user-images.githubusercontent.com/18424138/124802503-09bc0380-df26-11eb-91a8-ce12af52985d.png)
![image](https://user-images.githubusercontent.com/18424138/124803868-a337e500-df27-11eb-871d-9f2d4aad81dc.png)
![image](https://user-images.githubusercontent.com/18424138/124805260-476e5b80-df29-11eb-88d8-0212be4eda92.png)


- [ ] I know we are waiting on an API change for this researcher profile url, just want to make sure it will come through here once hooked up.
![image](https://user-images.githubusercontent.com/18424138/124803014-949cfe00-df26-11eb-86d6-40e6e6aa8249.png)

**Comment by kcullion** [comment]
From here....https://github.com/icgc-argo/dac-ui/issues/175
For testing:

Thanks @anncatton ! This is looking really great. Here's some preliminary style feedback...I will now try testing different content in the application that is now saveable. 

- [x] we discussed the font changes over slack
- [x] fix checkbox svgs (will send files over slack)
- [x] uppercase for DACO
![image](https://user-images.githubusercontent.com/18424138/124967219-2aea2600-dff2-11eb-87a7-86454d57bed2.png)

- [x] footer: `using` instead of `by`
![image](https://user-images.githubusercontent.com/18424138/124967307-448b6d80-dff2-11eb-9285-b4de4dbc4c00.png)

**Comment by kcullion** [comment]
@anncatton looks good! Tested on July 9

- Colons are now present
- Footer url fixed
- Lay summary being displayed
- Blank publications are shown on draft when no values have been added
- I now see the researcher profile url being displayed
- I see full name for applicant, IR and collaborator
- Ethics checkbox are now blank if I haven't checked them off
- Ethics first checkbox is checked off when I submit "false" in postman
- I see more space above the Use for Data in section D
- Font changes look great, now see signatures on one page
- Checkbox svgs look good
- Uppercase for DACO is fixed on title page

**Comment by rosibaj** [comment]
looks great! thank you @anncatton !
